lib/classes/Variables.test.js

```diff
@@ -156,6 +158,30 @@ describe('Variables', () => {
             .then(() => expect(property.getter(awsProvider)).to.be.eql('default'));
         });
       });
+
+      it('should not populate variables containing *', () => {
+        const property = { name: 'region', getter: (provider) => provider.getRegion() };
+
+        awsProvider.options[property.name] = '${self:foobar, *.a.thing}';
+        serverless.variables.populateService().should.be.fulfilled
+          .then(() => expect(property.getter(awsProvider)).to.be.eql('${self:foobar, *.a.thing}'));
+      });
+
+      it('should populate variables containing * if it is sourrounded by \'', () => {
+        const property = { name: 'region', getter: (provider) => provider.getRegion() };
+
+        awsProvider.options[property.name] = '${self:foobar, "*.a.thing"}';
```

Comment:
Minor nit: you seem to have the single and double quote cases reversed in name or implementation.